shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1272,43 +1277,67 @@ - (void)updateViewportPadding {
 
 #pragma mark - Keyboard events
 
-- (void)keyboardWillChangeFrame:(NSNotification*)notification {
+- (void)keyboardWillShowNotification:(NSNotification*)notification {
+  // Immediately prior to a docked keyboard being shown or when a keyboard goes from
+  // undocked/floating to docked, this notification is triggered
+
   NSDictionary* info = [notification userInfo];
+  CGRect keyboardFrame = [[info objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
```

Comment:
So changing it to that threw an error `property 'CGRectValue' not found on object of type 'id _Nullable'`. I think when it's chained like that it doesn't know what type it is. I could change it so it's like this and uses dot syntax, do you think this would be better?
```
NSValue* keyboardEndValue = info[UIKeyboardFrameEndUserInfoKey];
CGRect keyboardFrame = keyboardEndValue.CGRectValue;
```

Reply:
The local, or cast it, or keep the old bracket syntax (but use subscripting), up to you:
```objc
CGRect keyboardFrame = ((NSValue*)info[UIKeyboardFrameEndUserInfoKey]).CGRectValue;
```
```objc
[info[UIKeyboardFrameEndUserInfoKey] CGRectValue];
```